lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -882,7 +885,21 @@ struct InsertSliceAsyncOpConversion
                                  i32_val(byteWidth), i32_val(0));
           srcSize = ptxBuilder.newOperand(selectOp, "r");
         }
-        copyAsyncOp(dstOperand, srcOperand, copySize, srcSize);
+
+        // When 'other != 0' is supported, we will need to fold the op.getMask()
+        // and redundantDataMask() into the same predicate, the way it is done
+        // for LoadOp.
+        Value maskVal = redundantDataMask(srcTy, rewriter, loc);
+
+        // TODO: Masking does not work for CTA multicast with cp.async. This is
+        // a quick and dirty workaround to avoid the issue.
+        bool skipMaskForMultiCTA = triton::gpu::getNumCTAs(srcLayout) > 1;
```

Comment:
Can't we also hide this inside `redundantDataMask` and return `int_val(1, 1) whenever `numCTAs > 1` ?

Reply:
The thing is, for some reason this works just fine for regular loads, and I don't want to regress perf for them

Reply:
I see, makes sense.